x-pack/plugins/security_solution/public/detection_engine/rule_management/hooks/use_alert_suppression.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import type { Type } from '@kbn/securitysolution-io-ts-alerting-types';
+import { isSuppressibleAlertRule } from '../../../../common/detection_engine/utils';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+export interface UseAlertSuppressionReturn {
+  isSuppressionEnabled: boolean;
+}
+
+export const useAlertSuppression = (ruleType?: Type): UseAlertSuppressionReturn => {
+  const isThreatMatchRuleFFEnabled = useIsExperimentalFeatureEnabled(
+    'alertSuppressionForIndicatorMatchRuleEnabled'
+  );
+
+  const isSuppressionEnabledForRuleType = useCallback(() => {
+    if (!ruleType) return false;
+
+    if (ruleType === 'threat_match') return isThreatMatchRuleFFEnabled;
```

Comment:
The condition seems unnecessary since `if (ruleType === 'threat_match')` already checks the feature flag status for the suppressible `threat_match` rule. Once we remove the feature flag, we can eliminate this redundant condition.

If you have a specific scenario that is not covered please share it with me and I will add to a subsequent PR

Reply:
this can be bypassed, for example:

```
    if (ruleType === 'new_terms') return isNewTermsRuleFFEnabled;

```

but `new_terms` might bit be added to suppressible rule types

Reply:
[818eb8f](https://github.com/elastic/kibana/pull/175089/commits/818eb8f2f345d076db184701f432e502f30acd3e)